megamek/src/megamek/client/ui/swing/minimap/BoardviewlessMinimap.java

```diff
@@ -0,0 +1,563 @@
+/*
+ * Copyright (c) 2025 - The MegaMek Team. All Rights Reserved.
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License as published by the Free
+ * Software Foundation; either version 2 of the License, or (at your option)
+ * any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY
+ * or FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License
+ * for more details.
+ *
+ */
+
+package megamek.client.ui.swing.minimap;
+
+import megamek.client.IClient;
+import megamek.client.ui.swing.overlay.OverlayPainter;
+import megamek.client.ui.swing.overlay.OverlayPanel;
+import megamek.codeUtilities.MathUtility;
+import megamek.common.*;
+import megamek.common.actions.AttackAction;
+import megamek.common.actions.EntityAction;
+import megamek.common.enums.GamePhase;
+import megamek.common.event.*;
+
+import javax.swing.*;
+import java.awt.*;
+import java.awt.event.MouseAdapter;
+import java.awt.event.MouseEvent;
+import java.awt.event.MouseMotionAdapter;
+import java.awt.image.BufferedImage;
+import java.util.ArrayList;
+import java.util.List;
+
+public class BoardviewlessMinimap extends JPanel implements OverlayPainter {
```

Comment:
I will most likely see how to use the current minimap instead, and make it possible to change its graphics/style on demand... probably will only see this tomorrow.